### PR TITLE
Use `MakeWebGLSurface` in `toImageSync`

### PR DESCRIPTION
**Pull request by hterkelsen**
Uses a WebGL surface in conjunction with `readPixels` to make an image from an `SkPicture`. This is needed since the picture may contain GL texture-backed images.

Fixes https://github.com/flutter/flutter/issues/103803

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/test/canvaskit/picture_test.dart [hunk 1]
```diff
@@ -102,6 +102,27 @@ void testMain() {
...
+      final ui.Picture picture = recorder.endRecording();
+      final ui.Image image = picture.toImageSync(10, 15);
+
+      expect(image.width, 10);
+      expect(image.height, 15);
+
+      final ByteData? data = await image.toByteData();
+      expect(data, isNotNull);
+      expect(data!.lengthInBytes, 10 * 15 * 4);
+      expect(data.buffer.asUint32List().any((int byte) => byte != 0), isTrue);
```
Comment (yjbanov) [inline lib/web_ui/test/canvaskit/picture_test.dart:124]: Bonus points if this is also a golden test

**Review by yjbanov — APPROVED** [review]

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 7.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 9.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 11.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 13.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 14.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 15.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 16.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 17.
View them at https://flutter-engine-gold.skia.org/cl/github/35276

**Comment by jonahwilliams** [comment]
I'm not sure the entire context for this fix, but note that this is a large performance regression in cases where it is not needed. In general, reading back from the GPU is a high latency operation and should be avoided at all costs

File: lib/web_ui/lib/src/engine/canvaskit/picture.dart [hunk 2]
```diff
@@ -97,12 +100,27 @@ class CkPicture extends ManagedSkiaObject<SkPicture> implements ui.Picture {
...
     skCanvas.drawPicture(skiaObject);
     final SkImage skImage = skSurface.makeImageSnapshot();
+    final SkImageInfo imageInfo = SkImageInfo(
+      alphaType: canvasKit.AlphaType.Premul,
+      colorType: canvasKit.ColorType.RGBA_8888,
+      colorSpace: SkColorSpaceSRGB,
+      width: width,
+      height: height,
+    );
+    final Uint8List pixels = skImage.readPixels(0, 0, imageInfo);
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/canvaskit/picture.dart:116]: Specifically this part

**Pull request merged**